pkg/reconciler/v1alpha1/labeler/labeler.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labeler
+
+import (
+	"context"
+	"time"
+
+	"github.com/knative/pkg/controller"
+	"github.com/knative/pkg/logging"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/knative/pkg/tracker"
+	servinginformers "github.com/knative/serving/pkg/client/informers/externalversions/serving/v1alpha1"
+	listers "github.com/knative/serving/pkg/client/listers/serving/v1alpha1"
+	"github.com/knative/serving/pkg/reconciler"
+)
+
+const (
+	controllerAgentName = "labeler-controller"
+)
+
+// Reconciler implements controller.Reconciler for Route resources.
+type Reconciler struct {
+	*reconciler.Base
+
+	// Listers index properties about resources
+	routeLister         listers.RouteLister
+	configurationLister listers.ConfigurationLister
+	revisionLister      listers.RevisionLister
+	tracker             tracker.Interface
+}
+
+// Check that our Reconciler implements controller.Reconciler
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController wraps a new instance of this Reconciler in a controller.
+func NewController(
+	opt reconciler.Options,
+	routeInformer servinginformers.RouteInformer,
+	configInformer servinginformers.ConfigurationInformer,
+	revisionInformer servinginformers.RevisionInformer,
+) *controller.Impl {
+
+	c := &Reconciler{
+		Base:                reconciler.NewBase(opt, controllerAgentName),
+		routeLister:         routeInformer.Lister(),
+		configurationLister: configInformer.Lister(),
+		revisionLister:      revisionInformer.Lister(),
+	}
+	impl := controller.NewImpl(c, c.Logger, "Labels")
+
+	c.Logger.Info("Setting up event handlers")
+	routeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    impl.Enqueue,
+		UpdateFunc: controller.PassNew(impl.Enqueue),
+		DeleteFunc: impl.Enqueue,
+	})
+
+	// TODO(mattmoor): We should have a ResyncPeriod in reconciler.Options
```

Comment:
:100:

Reply:
I'll follow up with this.  Route controller needs this too.